src/dsp_tools/commands/xml_validate/api_connection.py

```diff
@@ -0,0 +1,73 @@
+from dataclasses import dataclass
+from typing import Any
+from typing import cast
+
+import requests
+from loguru import logger
+from requests import ReadTimeout
+from requests import RequestException
+from requests import Response
+
+from dsp_tools.models.exceptions import InternalError
+from dsp_tools.models.exceptions import UserError
+
+
+@dataclass
+class OntologyConnection:
+    api_url: str
+    shortcode: str
```

Comment:
After understanding what the class does, I see an issue in terms of separation of concerns and keeping the abstraction level consistent in related code:

`OntologyConnection` exposes two methods `get(url, headers)` and `get_ontologies()`. The latter is a high level concept that would fit an `OntologyClient`, while the first is a low-level concept that is specific to the HTTP protocol and absolutely fits a `OntologyConnection`.  
It would be absolutely fair to say that you call the class `OntologyClient` and make `get()` private. But in terms of separation of concerns and in terms of testability, it might even be the "cleaner" way, to separate this into two classes, where `OntologyClient` has a field `OntologyConnection`. But to be honest, I'm not sure that's worth the effort - because then in principal the `OntologyConnection` should be a protocol, and the concrete implementation would have to be `ApiOntologyConnection` (which in theory is nice because then you could have a `FileOntologyConnection` that can read ontologies from turtle files, and that would be perfectly interchangeable within the client... but realistically, you'll never need that).

Reply:
I agree that with regard to separation and testability it would be cleaner to take the `get` out here. I did not because I don't want to create another connection monster that does everything. Additionally I only want to handle errors here that may happen in this particular situation. Which is why I did not use the `Connection` class. So if I separated it, the OntologyConnection that does the get request would only ever be used in this class. Do you think that is overkill?

Reply:
I think in this instance, the pragmatic thing would be to make `get()` private `_get()` to hide the low level implementations from the caller that should just act on the high level. But it doesn't matter much here, the class is small enough to not become too confusing